bindings/nodejs/lib/types/unlock_condition.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Address, AliasAddress } from './address';
+
+/**
+ * All of the unlock condition types.
+ */
+enum UnlockConditionType {
+    Address = 0,
+    StorageDepositReturn = 1,
+    Timelock = 2,
+    Expiration = 3,
+    StateControllerAddress = 4,
+    GovernorAddress = 5,
+    ImmutableAliasAddress = 6,
+}
+
+abstract class UnlockCondition {
+    private type: UnlockConditionType;
+
+    constructor(type: UnlockConditionType) {
+        this.type = type;
+    }
+    /**
+     * The type of unlock condition.
+     */
+    getType(): UnlockConditionType {
+        return this.type;
+    }
+}
+
+class AddressUnlockCondition extends UnlockCondition /*implements IAddressUnlockCondition*/ {
```

Comment:
What are the comments for?

Reply:
ideally later we make it possible to implement the old interfaces still? or we dont care about them ever again?